service/watcher/chartstatus/kubeconfig.go

```diff
@@ -0,0 +1,114 @@
+package chartstatus
+
+import (
+	"context"
+	"time"
+
+	"github.com/giantswarm/apiextensions/v3/pkg/clientset/versioned"
+	"github.com/giantswarm/app/v4/pkg/key"
+	"github.com/giantswarm/backoff"
+	"github.com/giantswarm/errors/tenant"
+	"github.com/giantswarm/kubeconfig/v4"
+	"github.com/giantswarm/microerror"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+
+	"github.com/giantswarm/app-operator/v3/pkg/label"
+)
+
+// getG8sClient returns a versioned clientset for the kubeconfig used by the
+// chart-operator app CR running in the same namespace as the operator.
+func (c *ChartStatusWatcher) getG8sClient(ctx context.Context) (versioned.Interface, error) {
+	lo := metav1.ListOptions{
+		LabelSelector: label.ChartOperatorAppSelector(c.uniqueApp),
+	}
+	apps, err := c.k8sClient.G8sClient().ApplicationV1alpha1().Apps(c.appNamespace).List(ctx, lo)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+	if len(apps.Items) != 1 {
+		return nil, microerror.Maskf(executionFailedError, "expected 1 chart-operator app CR got %d", len(apps.Items))
+	}
+
+	// We have the chart-operator app CR. Now we get its kubeconfig.
+	app := apps.Items[0]
+
+	// App CR uses inCluster so we can reuse the existing client.
+	if key.InCluster(app) {
```

Comment:
Instead of chart-operator app CRs, can we use `unique` parameter instead to distinguish g8sclients for MC, WC?

Reply:
Yes I think that's better thanks.

If unique is true we can use the InCluster connection. If its false we use the namespace to generate the kubeconfig name,